torch/_inductor/codegen/triton.py

```diff
@@ -1136,9 +1136,27 @@ def __init__(self, line, var, mask, size_map):
             def __call__(self):
                 # The conditions need to be in parens because of Python's operator precedence.
                 # It'd be less # error-prone to use and/or/not, which is suported by triton
-                size = self.size_map[(self.var, self.mask)]
-                cond = f"(0 <= {self.var}) & ({self.var} < {size})"
-                cond_print = f"0 <= {self.var} < {size}"
+                size, size_str = self.size_map[(self.var, self.mask)]
+
+                assert_min = var.bounds.lower < 0
+                assert_max = var.bounds.upper > size
```

Comment:
Does this work when `size` is an expression?

Reply:
nah, in the dynamic shapes case we should always generate asserts. I have been working on getting the other three PRs to work. Now those are passing CI, I'll have a look at this one